app/components/OutcomeBar.tsx

```diff
@@ -1,64 +1,136 @@
-'use client';
-
-import { FixedProductMarketMaker } from '@/queries/omen';
 import { cx } from 'class-variance-authority';
+import { useQuery } from '@tanstack/react-query';
+import request from 'graphql-request';
+
+import { BlockDataType } from '@/queries/xdai';
+import {
+  FixedProductMarketMaker,
+  FpmmTrade_OrderBy,
+  getMarketTrades,
+} from '@/queries/omen';
+import { OMEN_SUBGRAPH_URL, XDAI_BLOCKS_SUBGRAPH_URL } from '@/constants';
 import { Outcome } from '@/entities';
 
 interface OutcomeBarProps {
   market: FixedProductMarketMaker;
 }
 
+type OutcomeTokenMarginalPricesResponse = {
+  [key: string]: {
+    outcomeTokenMarginalPrices: string[];
+  };
+};
+
 export const OutcomeBar = ({ market }: OutcomeBarProps) => {
+  const { id } = market;
+
+  const { data: trades } = useQuery({
+    queryKey: ['getMarketTrades', id],
+    queryFn: async () =>
+      getMarketTrades({
+        fpmm: id,
+        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
+      }),
+  });
+
+  const { data: lastTradeMarginalPrices } = useQuery({
+    queryKey: ['getLastTradeMarginalPrices', id],
+    queryFn: async (): Promise<string[] | undefined> => {
+      if (!trades) return;
+
+      const lastTradeTimestamp =
+        trades.fpmmTrades[trades.fpmmTrades.length - 1]?.creationTimestamp;
+
+      const blockNumber = await request<BlockDataType>(
+        XDAI_BLOCKS_SUBGRAPH_URL,
+        `query blockNumberByTimestamp {
+          _${lastTradeTimestamp}: blocks(where: {timestamp: ${lastTradeTimestamp}}, first: 1) {
+              number
+            }
+        }`
+      );
+
+      const tradeBlockNumber = blockNumber[`_${lastTradeTimestamp}`][0].number;
+
+      if (!tradeBlockNumber) return;
+
+      const marginalPricesResponse = await request<OutcomeTokenMarginalPricesResponse>(
+        OMEN_SUBGRAPH_URL,
+        `query marginalPricesByBlockNumber($id: ID!) {
+         _${tradeBlockNumber}: fixedProductMarketMaker(id: $id, block: { number: ${tradeBlockNumber} }) {
+              outcomeTokenMarginalPrices
+            }
+        }`,
+        {
+          id,
+        }
+      );
+
+      return marginalPricesResponse[`_${tradeBlockNumber}`].outcomeTokenMarginalPrices;
+    },
+    enabled: !!trades,
+  });
+
   const outcome0 = new Outcome(
     0,
     market.outcomes?.[0] || 'Option 1',
-    market.id,
-    market.outcomeTokenMarginalPrices?.[0],
-    market.question?.currentAnswer
+    id,
+    market.outcomeTokenMarginalPrices?.[0] ?? lastTradeMarginalPrices?.[0]
   );
   const outcome1 = new Outcome(
     1,
     market.outcomes?.[1] || 'Option 2',
-    market.id,
-    market.outcomeTokenMarginalPrices?.[1],
-    market.question?.currentAnswer
+    id,
+    market.outcomeTokenMarginalPrices?.[1] ?? lastTradeMarginalPrices?.[1]
   );
 
+  const hasPercentages = outcome0.percentage && outcome1.percentage;
```

Comment:
is there outcomes without percentage or are is it always a temporary state?

Reply:
The `ShimmerBar` is more like a loading state in general. I called it "Shimmer" due to the effect (it's not a spinner animation or a loading animation), I don't mind renaming it as "LoadingBar" or something like that if it's easier to read and understand.

We should always get a percentage response. We'll try to get it out the `market` data first, and if there's no market percentages data, we try to get it from the subgraph (first from XDAI explorer to get the block number, then from Omen with that block number).

I tried to use the `fetching` value out the `lastTrade` query data, the thing is that for many states we're using the `market` data and not the `lastTrade`, so we would force a not needed loading state (because we already have that data).

That's why I decided to use these two variables percentages, in a sense of "ok let's the outcome class handle the decision and we'll use the fastest response from that class"

Reply:
Renaming `ShimmerBar` to `LoadingOutcomeBar`, makes it clear on what it is. 

I think It's not that relevant what kind of loading it is, if a spinner or shimmer but that is a loading.

Related to the `hasOutcomePercentages`, I'm wondering if would be more clear if we did something like this

```js
const isLoading = !hasPercentages
```
And the we use the `isLoading` variable